Sources/Common/Core/DataArray/index.d.ts

```diff
@@ -64,71 +75,143 @@ export interface vtkDataArray extends vtkObject {
 	setRange(rangeValue: vtkRange, componentIndex: number): [number, number];
 
 	/**
-	 *
+	 * Set the given tuple at the given index.
 	 * @param {Number} idx
 	 * @param {Number[]} tuple
 	 */
 	setTuple(idx: number, tuple: number[]): void;
 
 	/**
-	 *
+	 * Get the tuple at the given index.
 	 * @param {Number} idx
 	 * @param {Number[]} [tupleToFill] (default [])
 	 */
 	getTuple(idx: number, tupleToFill?: number[]): number[];
 
+	/**
+	 * Insert the given tuple at the given index.
+	 * NOTE: May resize the data values array. "Safe" version of setTuple.
+	 *
+	 * A typical usage is when the `vtkDataArray` is initialized with
+	 * `initialValues = { size:0, values: new Uint8Array(1000) }`, where an
+	 * empty but pre-allocated array with 1'000 components is created.
+	 * The component values can then be inserted with `insertTuple()`
+	 * or `insertNextTuple()` without requiring new memory allocation until
+	 * the size of 1'000 is exceeded.
```

Comment:
the size of 1000 components is exceeded (e.g. when inserting the 250th 4-component-tuple)